### DISCUSSION TITLE
HmIP-BBL: Setting cover position and tilt position at the same time is buggy

### DISCUSSION THREAD

**Discussion by sleiner** [SukramJ/aiohomematic#1506] [Ideas]
### The problem

When changing both cover position ("Behanghöhe") and tilt ("Lamellenposition") at roughly the same time through the Home Assistant `cover.set_cover_position` and `cover.set_cover_tilt_position, the blind stutters for a bit and either the new position *or* the new tilt are applied, but not both.

#### How to reproduce

1. Set the blind blind position to cover position ("Behanghöhe") 10% and tilt ("Lamellenposition") 10%, via one of these methods:
   - CCU Web UI
   - Set cover position in the Home Assistant dashboard, wait until the blind has arrived, then set tilt position
   - Via the `homematicip_local.set_device_value`, set `COMBINED_PARAMETER` to `L=10,L2=10`
2. Create a Home Assistant scene which sets both cover position and tilt to 20%.
3. Activate the scene.

**Expected**: Blind applies a cover position of 20% and a tilt of 20%
**Actual behaviour:* Blind stutters a bit and applies a cover position of **10%** and a tilt of 20%.

#### Potential cause for the bug

When the scene is activated, Home Assistant calls `cover.set_position` and `cover.set_tilt_position` in rapid succession. After checking the cover implementation in `hahomematic`, I suppose that this results in two API calls to the CCU and thus two calls to `CeIpBlind._set_level()`. Even when the `level` or `tilt` level parameters are `None`, they are [defaulted to the current position](https://github.com/danielperna84/hahomematic/blob/f278a2de8a78938b57fd1303b4a07c789678df15/hahomematic/platforms/custom/cover.py#L282) so that combined parameter is likely set twice with these values:

1. `L=20,L2=10` (resulting from `cover.set_position`)
2. `L=10,L2=20` (resulting from `cover.set_tilt_position`)

Ideally, setting them like this _should_ work:

1. `L=20`
2. `L2=20`

Unfortunately, it does not - as I have tried using some CCU programs (without any Home Assistant involvement at all).
Likewise, this does also not work:

1. `L=20,L1=101`
2. `L=101,L2=20`

The only variant I found that actually worked was:

1. `L=20` (setting the position when the blind is not moving. As the blind is in a steady, setting only either `L` or `L2` is fine.
2. `L=20,L2=20` (setting the tilt position while the blind is already moving - note that we explicitly need to mention the target `L` even though it is not yet reached)

**Comment by SukramJ** [comment]
Ich kannst den Service [set_cover_combined_position](https://github.com/danielperna84/custom_homematic#homematicip_localset_cover_combined_position) verwenden, da von Haus aus keine kombinierte Parameterübertragung unterstützt.

Eine weitere Verbesserung wurde [hier](https://github.com/danielperna84/hahomematic/discussions/1269) vorgeschlagen, aber da vom Benutzer keine Rückmeldung mehr kam hab ich das Theme nicht weiter verfolgt. Vielleicht ist was Relevantes dabei.

Da ich selber keine `cover` besitze bin ich auf entsprechende Zuarbeit angewiesen.

> Ideally, setting them like this should work:

`Should` ist mir egal. 
Bitte teste das mit dem combined_parameter in der CCU und melde eine funktionierende Variante zurück.
`cover` ist mir zu sensibel, und wird von zu vielen Benutzern verwendet, als das ich an der Stelle rumexperimentieren werde.

**Comment by SukramJ** [comment]
Vielleicht auch hilfreich: #560

**Comment by sleiner** [comment]
Hi @SukramJ,

vielen Dank für die superschnelle Antwort!

> Ich kannst den Service [set_cover_combined_position](https://github.com/danielperna84/custom_homematic#homematicip_localset_cover_combined_position) verwenden, da von Haus aus keine kombinierte Parameterübertragung unterstützt.

Danke für den Hinweis! :-) Das ist mir bewusst; für Automatisierungen oder Skripte würde ich das sowieso bevorzugen, schon allein um durch nur einen Funkbefehl den Duty Cycle niedrig zu halten.
Ich fände es trotzdem erstrebenswert, die "Home Assistant native" Variante zum Laufen zu bringen, da in manchen Bereichen nur die nativen Services verwendet werden können und nicht die Homematic-spezifischen, zum Beispiel:
- Dashboard
- Szenen

>> Ideally, setting them like this should work:
>>
> `Should` ist mir egal.
> Bitte teste das mit dem combined_parameter in der CCU und melde eine funktionierende Variante zurück.

Das habe ich getan, es hat nicht funktioniert. Genauso die Variante aus #1269. Das hatte ich in meinem ursprünglichen Report wohl nicht gut formuliert. Um es nochmal anders und hoffentlich klarer zu formulieren:

Wenn gerade `cover.set_position` mit einer Position von 20 aufgerufen wurde und die Jalousie noch fährt, könnte `cover.set_tilt_position` für eine Lamellenposition von 20 den combined parameter folgendermaßen setzen:

- `L2=20`: setzt die Lamellenposition auf 20, die Behanghöhe kehrt auf den vorigen Wert zurück
- `L=101,L2=20`: setzt die Lamellenposition auf 20, die Jalousie fährt ganz nach oben, der Status in der CCU zeigt allerdings Lamellenposition 20% und Behanghöhe 20% an.
- `L=20,L2=20` ist das einzige, was mir einfällt, was funktioniert. Für `L` muss man dann aber halt den Sollwert aus dem letzten Service-Call angeben und nicht die aktuelle Position. Das wäre etwas mehr Komplexität in der Implementierung von `hahomematic`, würde aber das Problem lösen.

Wenn du möchtest, kann ich einen Pull Request vorbereiten. Oder hast du vielleicht noch eine andere Idee?

**Comment by SukramJ** [comment]
> Ich fände es trotzdem erstrebenswert, die "Home Assistant native" Variante zum Laufen zu bringen

Viel Erfolg. Sauber formuliert könntest Du natürlich auch [hier](https://github.com/home-assistant/architecture) sinnvolle Änderungen in HA erwirken.

> Wenn du möchtest, kann ich einen Pull Request vorbereiten. 

Bitte gerne, hier muss ich aber auf entsprechend angepasste bzw. erweiterte Unittests bestehen, da ich das ganze ansonsten später, aufgrund fehlender eigener Geräte, nicht weiter pflegen kann.

Alternativ wird ein entsprechender und möglichst vollständiger Testplan benötigt, und um die Implementierung würde ich mich kümmern
Meine Erwartung wäre hier, das alle relevanten Parameterkombinationen gegen die CCU getestet werden (kein `should`).
Parameter -> erwartetes Ergebnis.
Das Testen muss über einen `combined_parameter`, oder wie der Parameter auch in der CCU WebUI heißt, erfolgen.

> Sollwert aus dem letzten Service-Call angeben und nicht die aktuelle Position

Das hört sich ganz schlecht an, und sowas würde ich glaube ich auch nicht aktzeptieren. Was ist denn, wenn von einer anderen App aus geändert wird? 
Die Zustandsänderung sollte immer vom aktuellen Zustand ableitbar sein.
Die Lösung sollte so ein fach wie möglich sein, denn ich muss das später weiter pflegen!!!

**Reply by sleiner** [reply]
>> Sollwert aus dem letzten Service-Call angeben und nicht die aktuelle Position
>>
> Das hört sich ganz schlecht an, und sowas würde ich glaube ich auch nicht aktzeptieren. Was ist denn, wenn von einer anderen App aus geändert wird?
> Die Zustandsänderung sollte immer vom aktuellen Zustand ableitbar sein.

Prinzipiell sähe meine Idee so aus:

Statt dem aktuellen
```python
        _level = level if level is not None else self.current_position / 100.0
        _tilt_level = tilt_level if tilt_level is not None else self.current_tilt_position / 100.0
```
das pro Wert zwischen zwei Fällen unterscheidet, würde man zwischen drei Fällen unterscheiden:

```python
    if level is not None: # Level soll explizit gesetzt werden
        _level = level
    elif self.is_moving() and self.current_target_level is not None:
        # Die Jalousie bewegt sich und die Zielbehanghöhe ist bekannt
        _level = self.target_position
    else:  # Jalousie ist im Stillstand und kein Level explizit gewünscht => wir bleiben auf dem aktuellen Level
        _level = self.current_position / 100.0

    if tilt_level is not None: # Tilt soll explizit gesetzt werden
        _tilt_level = tilt_level
    elif self.is_moving() and self.current_target_level is not None:
        # Die Jalousie bewegt sich und die Ziellamellenposition ist bekannt
        _tilt_level = self.target_tilt_position
    else:  # Jalousie ist im Stillstand und kein Tilt explizit gewünscht => wir bleiben auf dem aktuellen Winkel
        _tilt_level = self.current_tilt_position / 100.0

    self.target_position = _level
    self.target_tilt_position = _tilt_level
```

Sobald sich die Jalousie nicht mehr bewegt aufhört, sich zu bewegen (ich sehe gerade nicht, wo es ein Callback dafür gibt, aber es muss eins geben, schließlich sehe ich es ja im Home Assistant Dashboard), würde man `target_position` und `target_tilt_position` wieder auf `None` setzen.

Um auf die Frage zurückzukommen:

> Was ist denn, wenn von einer anderen App aus geändert wird?

- Wenn eine andere App Behanghöhe oder Lamellenposition  **und die Jalousie noch fährt**, würden wir das überschreiben. Das tun wir aber auch jetzt schon. Und das tun auch die CCU Web-UI und [pocket control HM](https://penzler.de/phAnleitung/Anleitung/Haus.html) Stand jetzt genau so. Wir werden also nicht schlechter. Wir werden aber in dem Fall, dass aus Home Assistant mehrere Befehle hintereinander kommen, besser.
- Wenn eine andere App Behanghöhe oder Lamellenposition ändert **und die Jalousie fertig gefahren ist**, überschreiben wir natürlich nichts.

Soweit ich es sehe, geben es die aktuellen CCU-Schnittstellen nicht her, Behanghöhe und Lamellenposition tatsächlich unabhängig voneinander zu verändern *während die Jalousie fährt*. Entsprechend müsste `hahomematic` als Client der API selbst tracken, was es zuletzt requested hat, zumindest bis die Jalousie zum Stehen kommt.

> Die Lösung sollte so ein fach wie möglich sein, denn ich muss das später weiter pflegen!!!

So wie ich es beschrieben habe, wäre es für dich akzeptabel oder zu kompliziert um es zu maintainen? Vielen Dank auf jeden Fall schonmal für die detaillierte technische Diskussion :-)

**Reply by SukramJ** [reply]
Die `target_* `Werte sollen die Werte aus dem vorherigen ServiceCall sein?

> Soweit ich es sehe, geben es die aktuellen CCU-Schnittstellen nicht her, Behanghöhe und Lamellenposition tatsächlich unabhängig voneinander zu verändern während die Jalousie fährt.

Hast Du das auch mal mit dem in #1269 genannten Platzhalter für NO_MODIFICATION (101 bzw. 1.01) getestet?
Also den entsprechenden und nicht zu verändernden Wert mit 101 befüllen.

**Reply by sleiner** [reply]
>> Soweit ich es sehe, geben es die aktuellen CCU-Schnittstellen nicht her, Behanghöhe und Lamellenposition tatsächlich unabhängig voneinander zu verändern während die Jalousie fährt.
>>
> Hast Du das auch mal mit dem in https://github.com/danielperna84/hahomematic/discussions/1269 genannten Platzhalter für NO_MODIFICATION (101 bzw. 1.01) getestet?
> Also den entsprechenden und nicht zu verändernden Wert mit 101 befüllen.

Habe ich probiert, ohne Erfolg. Siehe [meine Nachricht vom Sonntag](https://github.com/danielperna84/hahomematic/discussions/1506#discussioncomment-9110370):

> Wenn gerade `cover.set_position` mit einer Position von 20 aufgerufen wurde und die Jalousie noch fährt, könnte `cover.set_tilt_position` für eine Lamellenposition von 20 den combined parameter folgendermaßen setzen:
>
> - `L2=20`: setzt die Lamellenposition auf 20, die Behanghöhe kehrt auf den vorigen Wert zurück
> - `L=101,L2=20`: setzt die Lamellenposition auf 20, die Jalousie fährt ganz nach oben, der Status in der CCU zeigt allerdings Lamellenposition 20% und Behanghöhe 20% an.
> - `L=20,L2=20` ist das einzige, was mir einfällt, was funktioniert. Für `L` muss man dann aber halt den Sollwert aus dem letzten Service-Call angeben und nicht die aktuelle Position. Das wäre etwas mehr Komplexität in der Implementierung von `hahomematic`, würde aber das Problem lösen.

In der Zwischenzeit hab ich noch probiert, was passiert, wenn ich direkt in die Datenpunkte für Behanghöhe und Lamellenposition schreibe (anstatt den `COMBINED_PARAMETER`) zu verwenden. Diese Calls haben keine Auswirkung, solange die Jalousie aufgrund einer vorherigen Aktion noch fährt.

Von daher sehe ich in der Tat keine Möglichkeit, Behanghöhe und Lamellenposition tatsächlich unabhängig voneinander zu verändern während die Jalousie fährt.

---

> Die `target_*` Werte sollen die Werte aus dem vorherigen ServiceCall sein?

Genau 👍🏻

**Reply by SukramJ** [reply]
Siehe PR #1512

**Reply by SukramJ** [reply]
Installier dir mal Beta 1.60.0.alt1.
In HACS bei der Integration Beta akivieren und dann explizit 1.60.0.alt1 herunterladen.

`target_level`/`target_tilt_level` werden beim Servicecall gesichert, und stehen für max. 60 Sekunden zur Verfügung.
Während der Bewegung sendet die CCU die Änderungen bis die Zielposition erreicht wurde. Dann wird der `target_`Wert gelöscht

**Reply by sleiner** [reply]
Sehr cool, vielen Dank! Ich hab es ausprobiert, leider ist das Verhalten noch wie zuvor. Also hab ich mal den Debugger angeschmissen:

- Der neue Branch für "Jalousie fährt und target (tilt) level ist bekannt" wirdnie genommen
- Das liegt daran, dass `CeBlind._target_level` and `CeBlind._target_tilt_level` immer `None` sind
- Das wiederum kommt vermutlich daher, dass aus dem Command Cache (via `CommandCache.get_last_value_send()`) nach einem Cache-Eintrag für den `LEVEL`-Parameter gesucht wird, der Cache enthält aber nur einen Eintrag für den `COMBINED_PARAMETER`-Parameter.

**Reply by SukramJ** [reply]
Mist, das habe ich verdrängt. Danke.

**Reply by SukramJ** [reply]
Dann versuch mal 1.60.0.alt2

**Reply by SukramJ** [reply]
[1.60.0.alt3](https://github.com/danielperna84/custom_homematic/releases/tag/1.60.0.alt3)

**Reply by sleiner** [reply]
Hi @SukramJ,

die 1.60.0.alt3 hat noch einen Bug: Im neuen "die Jalousie fährt gerade Branch" sind `level` und `tilt` jeweils Faktor 100 zu hoch 😅 Das hab ich bei mir lokal schnell korrigiert und jetzt verhält sich das Fahren der Jalousien übers Dashboard so wie erwartet... teilweise 🙈 In meinen Vorabtests hatte ich immer gesehen, dass das gleichzeitige Setzen von `L` und `L2` (auf konkrete Werte, also zwischen 0 und 100) dazu führt, dass diese Wertkombination angefahren wird. Was ich in meinen Vorabtests **nicht** bemerkt hatte: Das funktioniert nicht immer, sondern nur, wenn die neue Wertkombination so ist, dass der Motor zum Anfahren sofort nochmal umdrehen muss. Wenn dem nicht so ist, werden Werte teils ignoriert oder `L2` auf 0 gesetzt. Beispiele:

### Beispiel 1

**Ausgangsposition:** Jalousie steht auf 50% Behanghöhe, 50% Lamellenposition

- Wähle 80% Behanghöhe aus -> `L2=50,L=80` wird gesendet
- Wähle 70% Behanghöhe aus -> `L2=50,L=70` wird gesendet

**Ergebnis:** Jalousie fährt auf 70% Behanghöhe, **0**% Lamellenposition 🤷🏼‍♂️ 

### Beispiel 2

**Ausgangsposition:** Jalousie steht auf 50% Behanghöhe, 50% Lamellenposition

- Wähle 40% Behanghöhe aus -> `L2=50,L=40` wird gesendet
- Wähle 70% Lamellenposition aus -> `L2=70,L=40` wird gesendet

**Ergebnis:** Jalousie fährt auf 40% Behanghöhe, **0**% Lamellenposition 🤷🏼‍♂️ wo diese 0% herkommen, kann ich mir ehrlich gesagt nicht erklären

### Beispiel 3
**Ausgangsposition:** Jalousie steht auf 50% Behanghöhe, 0% Lamellenposition

- Wähle 40% Behanghöhe aus -> `L2=0,L=40` wird gesendet
- Wähle 50% Lamellenposition aus -> `L2=50,L=40` wird gesendet
- Wähle 60% Behanghöhe aus -> `L2=50,L=60` wird gesendet

**Ergebnis:** Jalousie fährt auf 60% Behanghöhe, 40% Lamellenposition 👍🏻

---

An dieser Stelle bin ich geneigt zu sagen, dass das Übermitteln neuer Werte während eine Jalousie fährt, CCU- bzw. aktorseitig so kaputt ist, dass man das von außen kaum ausgleichen kann. Es tut mir sehr leid, dass mir diese Fälle erst jetzt auffallen, nachdem die Implementierung schon quasi fertig ist - mit "mal eben zwei Schieber mit der Maus bewegen" geht das Testen natürlich deutlich einfacher und schneller von der Hand als mit "ein CCU-Programm pro Zielpunkt anlegen und diese dann in kreativer Reihenfolge aufrufen", das ist aber trotzdem eine schwache Entschuldigung 😖

Mir fällt jetzt auch kein (guter) Lösungsansatz für hahomematic mehr ein (oder hast du noch eine Idee?) - ansonsten werde ich mal an eQ-3 schreiben und anfragen, ob sie die Reaktion auf neue Zielpositionen, während die Jalousie fährt, nicht auf ihrer Seite lösen können.

**Reply by SukramJ** [reply]
Schade. 
Die letzte Idee wäre das zwischen den Kommandos jeweils ein Stop gesendet wird.
Das könntest Du ohne Codeänderung ausprobieren.

**Reply by sleiner** [reply]
Gute Idee 😊 Ich hab lokal den Code mal so geändert, dass (zumindest wenn die Jalousie fährt) vor dem `COMBINED_PARAMETER` noch ein Stop-Kommando gesendet wird. Wenn ich auf dem Dashboard in langsamen Tempo klicke, funktioniert es wie erwartet 👍🏻 Wenn die Service Calls allerdings zu schnell hinter einander kommen (z.B. wenn man im Dashboard Parameter in sehr schneller Reihenfolge ändert oder eine Szene aktiviert), laufen wir in eine von zwei Race Conditions:

1. Wenn von der CCU noch nicht die Rückmeldung kam, dass die Jalousie jetzt fährt, wird der neue Branch nie verwendet. Das Problem tritt in meinem Setup bei mehreren Service Calls innerhalb von ca. einer Sekunde auf.
3. Sehr viel kleineres Problem: `_set_level` kann natürlich auch in zwei Tasks nahezu gleichzeitig aufgerufen werden. Da die Funktion an einigen Stellen unterbrochen kann, könnte man auch da in Race Conditions laufen.

**Reply by SukramJ** [reply]
Und wie ist da Fazit?
Bietet die Änderung einen Mehrwert für den Benutzer?
Vielleicht macht es sollte man die Beding auf `moving` rausnehmen.
Der `target value` existiert, solange es keine Bestätigung durch die CCU gibt.
Als default für einen nicht gesetzten Parameter nimmt man also den `target value` sofern vorhanden, und ansonsten die aktuelle Position.

```
        _level = level if level is not None else (self._target_level or self._channel_level)
        _tilt_level = (
            tilt_level
            if tilt_level is not None
            else (self._target_tilt_level or self._channel_tilt_level)
        )
```

**Reply by sleiner** [reply]
> Und wie ist da Fazit?
> Bietet die Änderung einen Mehrwert für den Benutzer?

Wenn wir die Race Conditions gelöst bekommen, sehe ich da schon einen Mehrwert: Das "manuelle" Einstellen übers Dashboard würde funktionieren, und Szenen würden auch funktionieren. Stand jetzt passieren halt für Benutzer schwer erklärliche Dinge, wenn man versucht, Szenen überhaupt zu verwenden oder auf dem Dashboard Einstellungen zu justieren, während die Jalousie fährt. Das ist aus meiner Sicht ein klarer Gewinn, gerade auch für den Smart-Home-Akzeptanz-Faktor 😉

Der Nachteil ist allerdings, dass die Jalousie beim Nachjustieren in Bewegung immer kurz anhält und dass beim Nachjustieren in Bewegung immer zwei zusätzliche Kommandos gesendet werden. Ich persönlich werde (weiterhin) dort, wo es problemlos möglich ist (z.B. in Automatisierungen) `set_cover_combined_position` verwenden.

---

> Vielleicht macht es sollte man die Beding auf moving rausnehmen.
> Der `target value` existiert, solange es keine Bestätigung durch die CCU gibt.
> Als default für einen nicht gesetzten Parameter nimmt man also den `target value` sofern vorhanden, und ansonsten die aktuelle Position.

Klingt hervorragend 👍🏻 (zumindest, wenn du mit "Bestätigung" die Bestätigung meinst, dass die Jalousie jetzt an der Zielposition angekommen ist - man müsste den Callback darüber, dass die Jalousie sich jetzt bewegt, da halt ignorieren).

---

> ```python
>        _level = level if level is not None else (self._target_level or self._channel_level)
>        _tilt_level = (
>            tilt_level
>            if tilt_level is not None
>            else (self._target_tilt_level or self._channel_tilt_level)
>        )

Das ist vermutlich eh nur Beispielcode, aber als Hinweis, weil das so ein klassisches subtiles Problem ist: Der Code hier wäre für `self._target_tilt_level == 0` buggy, weil 0 ja auch falsy ist.

**Reply by SukramJ** [reply]
Da ich keine Zugriff auf so ein Gerät habe machen weitere Versuche für mich erst mal keinen Sinn mehr.
Es wäre sinnvoller, wenn Du entsprechende Tests bei dir lokal durchführst, und einen Pullrequest erstellst. 
Den werde ich mir natürlich gerne anschauen.

**Reply by sleiner** [reply]
Alles klar, ich bereite dieser Tage was vor :-)

**Reply by SukramJ** [reply]
Sehr gut.
Ich war in der letzten Zeit trotzdem nicht ganz untätig und hab ein paar Sachen eingebaut, die dir vielleicht helfen könnten:

Die von der CC aufgerufen Methoden habe die Annotation `@bind_collector,` du um zwei Parameter ergänzt wurde:

`wait_for_callback`:
Default ist `None` und somit deaktiviert.
Hier kann ein Timeout eingetragen werden wie lange die Methode auf einen einen CCU-Callback mit dem Zielwert warten soll.

`use_command_queue`:
Default ist `False` und somit deaktiviert. 
Wenn aktiviert, werden die abgesetzten Befehle in einer FIFO-Queue pro Gerät abgelegt, und der Reihe nach abgearbeitet.

Die aktivierte Kombination von `wait_for_callback` und `use_command_queue` hat zur Folge, das abgesetzte Befehle nacheinander und Vollständig abgearbeitet werden.
Ausgenommen hier ist der STOP Befehl, der sofort gesendet wird und die Gerätequeue leert.
Die entsprechenden Methoden wurden im PR #1512 bereits annotiert.

Ob das Ganze bei einem `HmIP-BBL` Sinn macht und funktioniert kann ich leider nicht sagen.
Vielleicht helfen die neuen Parameter ja.

**Reply by sleiner** [reply]
PR ist eröffnet: #1549

**Reply by SukramJ** [reply]
Mit Beta [1.62.0b0](https://github.com/danielperna84/custom_homematic/releases/tag/1.62.0b0) veröffentlicht

**Reply by sleiner** [reply]
Ich hab es eben erfolgreich auf meinem Produktivsystem getestet :-)

**Comment by martin-st-81** [comment]
Hallo, wünschen einen schönen Tag :-)
Wir haben 10 HMIP-BBL in Verwendung und diese schließen leider auch mit der Version 1.62.0b0 nicht mehr komplett. Sie schließen zu 100% und fahren dann auf Lamellenposition 50%. Was bedeutet, dass sich diese praktisch wieder öffnen und aus dem Jalousiekasten mehrere Zentimeter herausfahren und ungeschützt sind. Dieses Verhalten tritt bei direkter Ansteuerung aus HomeMatic nicht auf. 
Es werden die unveränderten Entitäten von Homematic(IP) Local - cover.hmip_bbl_XYZ in HomeAssistant 2024.5.4 verwendet.
Wie kann die Lamellenposition wieder allgemein beim schließen auf 0% gesetzt werden ?
Ich habe bei den HMIP-BBL bereits die Auto Calibration in der HomeMatic ausgeführt und das Homematic(IP) Local im HomeAssistant neu installiert.
Bitte um Unterstützung, Danke

**Reply by SukramJ** [reply]
Siehe #1552

**Reply by SukramJ** [reply]
Bitte in diesem thread dazu keine weitere Diskussion führen, das passt inhaltlich nicht.
